src/include/dns-server/dialog-masterzone.rb

```diff
@@ -207,37 +200,37 @@ def GetMasterZoneEditorTabBasics
       deep_copy(contents)
     end
 
-    def ZoneAclInit
-      allowed = false
-      keys = []
-      Builtins.foreach(Ops.get_list(@current_zone, "options", [])) do |m|
-        if Ops.get_string(m, "key", "") == "allow-transfer" && !allowed
-          key = Builtins.regexpsub(
-            Ops.get_string(m, "value", ""),
-            "^.*\\{[ \t]*(.*)[ \t]*\\}.*$",
-            "\\1"
-          )
+    # @return [Array<String>]
+    def current_zone_allow_transfer
+      items = []
+      @current_zone.fetch("options", []).each do |m|
+        if m["key"] == "allow-transfer"
+          key = m.fetch("value", "")
+          key = key[/\A.*\{[ \t]*(.*)[ \t]*\}.*\z/, 1]
           if key != nil
-            keys = Builtins.splitstring(key, " ;")
-            keys = Builtins.filter(keys) { |k| k != "" }
-            allowed = true
+            items = key.split(/[ \t;]/).find_all{|i| !i.empty?}
+            break
           end
         end
       end
+      items
```

Comment:
I think using each_with_object make this method shorter and more focused,,,and to be honest instead of break, it can use return that key.split

in general, I think that code work-flow can be more readable like

def current_zone_allow_transfer
  target_pair = @current_zone.fetch("options", []).find { |m| m["key"] == "allow-transfer" }

  return [] unless target_pair

  value = target_pair["value"]
  return [] unless value

  value.split(/[ \t;]/).reject(&:empty?)
end
```